web_src/js/features/repo-issue-content.js

```diff
@@ -13,20 +13,17 @@ function showContentHistoryDetail(issueBaseUrl, commentId, historyId, itemTitleH
 
   $dialog = $(`
 <div class="ui modal content-history-detail-dialog">
-  <i class="close icon inside"></i>
-  <div class="header">
-    ${itemTitleHtml}
-    <div class="ui dropdown right dialog-header-options" style="display: none; margin-right: 50px;">
-      ${i18nTextOptions} <i class="dropdown icon"></i>
+  ${svg('octicon-x', 16, 'close icon inside')}
+  <div class="header df ac sb">
+    <div>${itemTitleHtml}</div>
+    <div class="ui dropdown dialog-header-options df ac mr-5 hide">
+      ${i18nTextOptions}${svg('octicon-triangle-down', 14, 'dropdown icon')}
       <div class="menu">
         <div class="item red text" data-option-item="delete">${i18nTextDeleteFromHistory}</div>
       </div>
     </div>
   </div>
-  <!-- ".modal .content" style was polluted in "_base.less": "&.modal > .content"  -->
-  <div class="scrolling content" style="text-align: left; min-height: 30vh;">
```

Comment:
IMO The `min-height: 30vh` was done by purpose:
* to show the loading UI without flicker
* to make the dialog look better if there a only a few lines in the diff.

Reply:
Imho, we should just eliminate the loading state of the modal and fetch all data after click and only show the modal when all data is present.

I don't agree about superflous whitespace on small diffs 😉

Reply:
The user needs a UI feed back when they clicks the history menu item, so, something must be shown.


The dialog height in current approach can be very small, which makes the loading UI look strange.

Reply:
Could probably add a `is-loading` spinner onto the dropdown item. A bit unconventional but should work.

Reply:
The example was there: https://github.com/go-gitea/gitea/pull/21990#issuecomment-1335085467

Or you can reproduce by:

* open the history menu
* disconnect your network (or set emulate a laggy network)
* click a history item

Reply:
Yes that's the broken loading state, but if we don't have a loading state, this won't be an issue.